### PR TITLE
[incubator/corturn] New chart for coturn service on incubator

### PR DESCRIPTION
**Pull request by wolmi**
Signed-off-by: Miquel Ramon Ortega Tido <lenyos@gmail.com>

#### What this PR does / why we need it:
This PR is for adding a new chart on the incubator folder and starting with the process of going to the stable folder. The new chart is for the service [coturn](https://github.com/coturn/coturn), as defined on the [Readme.md](https://github.com/coturn/coturn/blob/master/README.md):

> The TURN Server is a VoIP media traffic NAT traversal server and gateway. It can be used as a general-purpose network traffic TURN server and gateway, too.

I want to collaborate by adding new services to be available as a chart.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

File: incubator/coturn/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1beta2
```
Comment (unguiculus) [inline incubator/coturn/templates/deployment.yaml]: Use `apps/v1`

File: incubator/coturn/values.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+ # Default values for coturn.
+ # This is a YAML-formatted file.
+ # Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: wolmi/coturn
+  tag: stable
```
Comment (unguiculus) [inline incubator/coturn/values.yaml]: Use a stable tag.

File: incubator/coturn/templates/config.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: {{ template "coturn.fullname" . }}-configmap
+  labels:
+    app: {{ template "coturn.name" . }}
+    chart: {{ template "coturn.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+    turnserver.conf: |+
+    {{ template "config_turnserver" .Values.config }}
```
Comment (unguiculus) [inline incubator/coturn/templates/config.yaml]: Please indent two spaces.

**Review by unguiculus — CHANGES_REQUESTED** [review]
I'd suggest you move the chart to stable.

File: incubator/coturn/values.yaml (hunk 2 above)
Comment (wolmi) [reply incubator/coturn/values.yaml]: Sorry, I don't understand dis comment, I'm now using a "stable" tag, do you want me to use some other format?

File: incubator/coturn/templates/deployment.yaml (hunk 1 above)
Comment (wolmi) [reply incubator/coturn/templates/deployment.yaml]: Done

File: incubator/coturn/templates/config.yaml (hunk 3 above)
Comment (wolmi) [reply incubator/coturn/templates/config.yaml]: Done

File: incubator/coturn/values.yaml (hunk 2 above)
Comment (unguiculus) [reply incubator/coturn/values.yaml]: Just because it is named `stable` doesn't mean it is stable. You should use a concrete version.

Comment (wolmi) [reply incubator/coturn/values.yaml]: Understand, I changed the image version to 0.1

File: incubator/coturn/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+{{- end -}}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "coturn.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "config_turnserver" -}}
```
Comment (unguiculus) [inline incubator/coturn/templates/_helpers.tpl]: All helper templates should be prefixed with `coturn`. Also, please use camel case.

`coturn.configTurnserver`

**Review by unguiculus — CHANGES_REQUESTED** [review]
I'd suggest you move the chart to stable and give it a `1.0.0`. We tend to move away from incubator here. Could you please update the chart to use the new label conventions?

See https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#names-and-labels

**Pull request closed without merging**